FirebaseCore/Internal/Sources/HeartbeatLogging/_ObjC_HeartbeatController.swift

```diff
@@ -49,10 +49,15 @@ public class _ObjC_HeartbeatController: NSObject {
   ///
   /// - Note: This API is thread-safe.
   /// - Returns: A heartbeats payload for the flushed heartbeat(s).
-  @available(iOS 13.0, macOS 10.15, macCatalyst 13.0, tvOS 13.0, watchOS 6.0, *)
-  public func flushAsync() async -> _ObjC_HeartbeatsPayload {
-    let heartbeatsPayload = await heartbeatController.flushAsync()
-    return _ObjC_HeartbeatsPayload(heartbeatsPayload)
+  public func flushAsync(completionHandler: @escaping (_ObjC_HeartbeatsPayload) -> Void) {
+    if #available(iOS 13.0, macOS 10.15, macCatalyst 13.0, tvOS 13.0, watchOS 6.0, *) {
+      Task {
+        let heartbeatsPayload = await heartbeatController.flushAsync()
```

Comment:
I forget if there is a learning related to this from the Auth rewrite.

Reply:
Looking at https://github.com/firebase/firebase-ios-sdk/pull/13521, it could be potentially risky that flushAsync() calls into GCD code.

Reply:
> Looking at #13521, it could be potentially risky that flushAsync() calls into GCD code.

It does not this, but I have an altermative approach to this PR in #13954 that does away with the internal async/await API we are wrapping at runtime here.

Reply:
A little background on when the highlighted code path will be called:
https://github.com/firebase/firebase-ios-sdk/blob/7a10b3bb3482b53720e71a69fd12f633c4531832/FirebaseAuth/Sources/Swift/Backend/AuthBackend.swift#L102

https://github.com/firebase/firebase-ios-sdk/blob/7a10b3bb3482b53720e71a69fd12f633c4531832/FirebaseCore/Extension/FIRHeartbeatLogger.h#L43-L44

  - _Xcode generated glue code creates the async/await version of the above API_

https://github.com/firebase/firebase-ios-sdk/blob/7a10b3bb3482b53720e71a69fd12f633c4531832/FirebaseCore/Sources/FIRHeartbeatLogger.m#L89-L94

which arrives at the highlighted implementation

Reply:
Resolving this as it's outdated. Let's go with implementation in https://github.com/firebase/firebase-ios-sdk/pull/13521